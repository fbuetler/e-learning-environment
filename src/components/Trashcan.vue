<template>
  <div
    id="trashcan"
    class="trashcan flex-item flex-center dropzone"
    title="Element entfernen!"
    @click="$emit('trashed-element', $event)"
    @dragover.prevent
    @dragenter.prevent
    @drop.stop.prevent="$emit('trashed-element', $event)"
  >
    <img :src="require('@/assets/icons/trashcan.png')" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";

@Component<Trashcan>({})
export default class Trashcan extends Vue {}
</script>

<style scoped>
.trashcan > img {
  width: 100%;
  max-width: 80px;
  margin: 1em;
}
</style>
