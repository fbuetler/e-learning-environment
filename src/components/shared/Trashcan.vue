<template>
  <div
    id="trashcan"
    class="flex-item flex-center dropzone responsive"
    title="Element entfernen!"
    @click="$emit('trashed-element', $event)"
    @dragover.prevent
    @dragenter.prevent
    @drop.stop.prevent="$emit('trashed-element', $event)"
  >
    <img :src="require('@/assets/icons/trashcan.png')" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";

@Component<Trashcan>({})
export default class Trashcan extends Vue {}
</script>

<style scoped>
.dropzone > img {
  margin: 1em;
}
</style>
