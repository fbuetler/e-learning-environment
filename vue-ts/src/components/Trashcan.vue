<template>
  <div
    class="trashcan"
    @click="$emit('trashed-element', $event)"
    @dragover.prevent
    @dragenter.prevent
    @drop.stop.prevent="$emit('trashed-element', $event)"
  >
    <img :src="require('@/assets/icons/trashcan.png')" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";

@Component<Trashcan>({})
export default class Trashcan extends Vue {}
</script>

<style scoped>
.trashcan {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 3px dashed black;
  margin: 0em 1em 0em 1em;
}
.trashcan > img {
  width: 100%;
  max-width: 80px;
  margin: 1em;
}
</style>
