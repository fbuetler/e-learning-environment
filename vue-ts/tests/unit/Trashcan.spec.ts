import { shallowMount } from "@vue/test-utils";
import Trashcan from "@/components/Trashcan.vue";

describe("Trashcan.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Trashcan);
  });

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("event triggered on click", async () => {
    await wrapper.find("#trashcan").trigger("click");
    expect(wrapper.emitted("trashed-element")).toBeTruthy();
  });

  it("event triggered on drop", async () => {
    await wrapper.find("#trashcan").trigger("drop");
    expect(wrapper.emitted("trashed-element")).toBeTruthy();
  });
});
