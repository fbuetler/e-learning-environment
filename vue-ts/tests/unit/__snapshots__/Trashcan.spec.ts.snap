// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trashcan.vue renders correctly 1`] = `
<div
  class="trashcan flex-item flex-center dropzone"
  id="trashcan"
  title="Element entfernen!"
>
  <img
    src=""
  />
</div>
`;
